datafusion/physical-expr/src/planner.rs

```diff
@@ -60,12 +59,9 @@ pub fn create_physical_expr(
         );
     }
     match e {
```

Comment:
Looking at this match statement, it seems like we could almost use `TreeNode::transform_up` and avoid the explicit recursion (not in this PR, I just noticed it while reviewing)